pages/index.tsx

```diff
@@ -1,23 +1,41 @@
+import React, { useState, useEffect } from 'react';
+
 import Head from 'next/head';
 import WhyKedro from '../components/why-kedro';
 import Header from '../components/header';
 import Hero from '../components/hero';
 import Features from '../components/features';
 import CaseStudies from '../components/case-studies';
-
-import style from '../styles/pages/index.module.scss';
+import Footer from '../components/footer';
 
 const Home = () => {
+  const [offset, setOffset] = useState(0);
+  const [showHeader, setShowHeader] = useState(true);
+
+  useEffect(() => {
+    const onScroll = () => setOffset(window.innerHeight + window.scrollY);
+    const footerOffset = document.getElementById('footer').offsetTop;
+    if (footerOffset <= offset) {
+      setShowHeader(false);
+    } else {
+      setShowHeader(true);
+    }
+    window.removeEventListener('scroll', onScroll);
+    window.addEventListener('scroll', onScroll, { passive: true });
+    return () => window.removeEventListener('scroll', onScroll);
+  });
```

Comment:
Good thinking here. We can improve on it though. This implementation causes the components to re-render each time the user scrolls. That's isn't great for performance.

I would try to implement [this instead](https://usehooks.com/useOnScreen/). We can create a top-level `/utils` folder. In that folder, I would create a `/hooks` folder, and then add this `useOnScreen.tsx` file with the hook inside it. Then you can use that hook in the `index.tsx` page.